client/ayon_traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -388,6 +390,54 @@ def _resolve_repre_path(
 
         return filepath
 
+    @staticmethod
+    def _validate_parents(project_name: str, product_item: ProductItem) -> list:
+        """ Ensure parent exists for provided product_item.folder_path
+
+        Args:
+            project_name (str): The project name.
+            product_item (ProductItem): The product item to inspect.
+
+        Returns:
+            list. The parent list if any
+
+        Raise:
+            ValueError: When provided folder_path parent do not exist.
+        """
+        parent_folder_path, folder_name = product_item.folder_path.rsplit("/", 1)
+        if not parent_folder_path:
+            return []
+
+        folder_data = ayon_api.get_folder_by_path(
+            project_name,
+            parent_folder_path,
+            fields=("folderType",)
+        )
+
+        if folder_data is None:
+            raise ValueError(f"Unknown parent folder {parent_folder_path}")
+
+        parent_data = []
+        inspected = []
+
+        for key in parent_folder_path.split("/"):
+            if not key:
+                continue
+
+            inspected.append(key)
+            folder_data = ayon_api.get_folder_by_path(
+                project_name,
+                f"/{'/'.join(inspected)}",
+                fields=("folderType",)
+            )
+            parent_data.append({
+                "folder_type": folder_data["folderType"],
+                "entity_name": key
+            })
+
+        return parent_data
```

Comment:
Simpified, with single call to server.
```suggestion
        parent_folder_names = product_item.folder_path.lstrip("/").split("/")
        # Rename name of folder itself
        parent_folder_names.pop(-1)
        if not parent_folder_names:
            return []
    
        parent_paths = []
        parent_path = ""
        for name in parent_folder_names:
            path = f"{parent_path}/{name}"
            parent_paths.append(path)
            parent_path = path
    
        folders_by_path = {
            folder["path"]: folder
            for folder in ayon_api.get_folders(
                project_name,
                folder_paths=parent_paths,
                fields={"folderType", "path"}
            )
        }
        parent_data = []
        for path in parent_paths:
            folder_entity = folders_by_path.get(path)
            if not folder_entity:
                raise ValueError(f"Unknown parent folder {path}")
            name = path.rsplit("/", 1)[-1]
            parent_data.append({
                "folder_type": folder_entity["folderType"],
                "entity_name": name
            })
        return parent_data
```